corehq/apps/api/tests/test_case_update_api.py

```diff
@@ -0,0 +1,179 @@
+
+
+class TestUpdateCases(TestCase):
+    domain = 'test_update_cases'
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.domain_obj = create_domain(cls.domain)
+        cls.web_user = WebUser.create(cls.domain, 'netflix', 'password', None, None)
+        cls.case_accessor = CaseAccessors(cls.domain)
+        cls.form_accessor = FormAccessors(cls.domain)
+
+    def tearDown(self):
+        FormProcessorTestUtils.delete_all_cases(self.domain)
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.domain_obj.delete()
+        super().tearDownClass()
+
+    def post(self, resource_url, body):
+        # TODO
+        pass
+
+    def test_create_case(self):
+        res = self.post('create_case', {
+            # notable exclusions: case_id, date_opened, date_modified
+            '@case_type': 'player',
+            '@case_name': 'Elizabeth Harmon',
+            '@owner_id': 'methuen_home',
+            'properties': {
+                'external_id': '1',
+                'sport': 'chess',
+                'dob': '1948-11-02',
+            },
+        })
+        self.assertEqual(res.keys(), ['@case_id', '@form_id'])
+        case = self.case_accessor.get_case(res['@case_id'])
+        self.assertEqual(case.domain, self.domain)
+        self.assertEqual(case.type, 'player')
+        self.assertEqual(case.name, 'Elizabeth Harmon')
+        self.assertEqual(case.owner_id, 'methuen_home')
+        self.assertEqual(case.opened_by, 'netflix')
+        self.assertEqual(case.external_id, '1')
+        self.assertEqual(case.dynamic_case_properties(), {
+            'dob': '1948-11-02',
+            'sport': 'chess',
+        })
+
+        xform = self.form_accessor.get_form(res['@form_id'])
+        self.assertEqual(xform.metadata.xmlns, 'http://commcarehq.org/case_api')
+        self.assertEqual(xform.metadata.userID, self.web_user.user_id)
+        self.assertEqual(xform.metadata.deviceID, 'user agent string')
+
+    def _make_case(self):
+        xform, cases = submit_case_blocks([CaseBlock(
+            case_id=str(uuid.uuid4()),
+            case_type='player',
+            case_name='Elizabeth Harmon',
+            owner_id='methuen_home',
+            create=True,
+            update={
+                'external_id': '1',
+                'sport': 'chess',
+                'rank': '1600',
+                'dob': '1948-11-02',
+            }
+        ).as_text()], domain=self.domain)
+        return cases[0]
+
+    def test_update_case(self):
+        case = self._make_case()
+
+        res = self.post(f'update_case/{case.case_id}', {
+            # notable exclusions: case_id, date_opened, date_modified, case_type
+            '@case_name': 'Beth Harmon',
+            '@owner_id': 'us_chess_federation',
+            'properties': {
+                'rank': '2100',
+                'champion': 'true',
+            },
+        })
+        self.assertEqual(res.keys(), ['@case_id', '@form_id'])
```

Comment:
See my comments below for bulk update. It might be interesting to return the whole case.